index.js

```diff
@@ -102,3 +109,43 @@ axios
     setFailed(error.message);
     endGroup();
   });
+
+let pipelineComplete = false;
+const pollInterval = 3000; // in milliseconds
+
+const keepPolling = getInput("Follow").toLowerCase();
+
+const pollWorkflow = () => {
+  axios
+    .get(workFlowUrl, {
+      headers: headers,
+    })
+    .then((response) => {
+      if (response.data.items[0].status != "running") {
+        pipelineComplete = true;
```

Comment:
According to [the CircleCI API](https://circleci.com/docs/api/v2/index.html#operation/getWorkflowById), possible statuses include:
> Enum: "success" "running" "not_run" "failed" "error" "failing" "on_hold" "canceled" "unauthorized" 

I found [this documentation for workflow states](https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/workflows.md), which also lists a "needs setup" step not documented in the API... 🤔

Based on the two of these, we should continue polling when states are one of the following:
* `running`
* `not_run` (Workflow was never started)
* `on_hold` (A job in the workflow is waiting for approval)

You should check for any status within that set.